client/src/partials/cash/cash.html

```diff
@@ -1,137 +1,217 @@
+                      </tr>
+                    </tbody>
+
+                    <!-- total rows -->
+                    <tfoot>
+
+                      <!-- only show the sum if we have more rows than 1 -->
+                      <tr ng-show="CashCtrl.payment.invoices.length > 1">
+                        <td colspan="2" class="text-right"><u>{{ "COLUMNS.TOTAL" | translate }}</u></td>
+                        <td class="text-right">{{ CashCtrl.slip.rawTotal | currency:CashCtrl.enterprise.currency_id }}</td>
+                      </tr>
+
+                      <!-- only show this segment if using a foreign currency -->
+                      <tr>
+                        <td colspan="2" class="text-right"><u>{{ "COLUMNS.EXCHANGE_RATE" | translate }}</u></td>
+                        <td class="text-right">
+                          <span ng-show="CashCtrl.payment.currency_id !== CashCtrl.enterprise.currency_id">
+                            {{ CashCtrl.slip.rate | currency:CashCtrl.payment.currency_id }}
+                          </span>
+                          <span ng-show="CashCtrl.payment.currency_id === CashCtrl.enterprise.currency_id">
+                            ---
+                          </span>
+                        </td>
+                      </tr>
+
+                      <!-- show a nice total row of what the payer owes today -->
+                      <tr>
+                        <td colspan="2" class="text-right"><u>{{ "COLUMNS.TOTAL" | translate }}</u></td>
+                        <td class="text-right">{{ CashCtrl.slip.total | currency:CashCtrl.payment.currency_id }}</td>
+                      </tr>
+                    </tfoot>
+                  </table>
+
+                  <button class="btn-block btn btn-warning" ng-click="CashCtrl.openInvoicesModal()" type="button">
+                    <span class="glyphicon glyphicon-plus-sign"></span> {{ "CASH.VOUCHER.SELECT_INVOICES" | translate }}
+                  </button>
+
+                  <!-- show informational message if there are not sales selected -->
+                  <p ng-if="CashCtrl.payment.invoices.length === 0" class="text-center text-primary">
+                    <span class="glyphicon glyphicon-info-sign"></span> {{ "CASH.VOUCHER.NO_INVOICES_SELECTED" | translate }}
+                  </p>
+                </div>
+
+                <hr />
+
+                <!--
+                  no matter what payment type, show the payment total details
+                -->
+                <div class="form-group" ng-class="{ 'has-error' : CashVoucherForm.$submitted && CashVoucherForm.amount.$invalid }">
+                  <label>{{ "COLUMNS.PAYMENT" | translate }}</label>
+                  <div class="input-group">
+                    <span class="input-group-addon">
+                      {{ CashCtrl.payment.currency_id ? CashCtrl.currencies.symbol(CashCtrl.payment.currency_id) : '...' }}
```

Comment:
This comment will not block the pull request. This method will be called on every angular digest, meaning if any unrelated input changes the whole statement will be re-evaluated (calling the function x times, often up to 10 times). 

Is it possible to move this calculation into a variable that is updated when the currency_id changes? The optimisation is very small in this case however it is considered good practice.